src/texstudio.cpp

```diff
@@ -1892,6 +1892,29 @@ LatexEditorView *Texstudio::editorViewForLabel(LatexDocument *doc, const QString
 	return qobject_cast<LatexEditorView *>(targetDoc->getEditorView());
 }
 
+LatexEditorView *Texstudio::editorViewForCommandDefinition(LatexDocument *doc, const QString &command)
+{
```

Comment:
Not sure if I understand your comment. These methods have to look up the definition of the given label/command/package in order to determine the corresponding editor view. I agree that there is some code duplication here, but I don't see how using tokens would help.

Reply:
the idea is that if you have basically the same method with only one line changed, you could use one method with a if/them selection on where to look for the needed info. And since the token tells you, it is a command, package,reference, one could just pass the token and let the method figure out the rest...

Reply:
I see your point. The case distinction on the token is already done in `Texstudio::editGotoDefinition` and I would prefer to have it in one place only. So I would propose to move the lookup to `editGotoDefinition` and then simply remove these methods (since they do not seem to be used from anywhere else).

If we do the same for `gotoToLabel` and `gotoToCommand` etc., we can also avoid the duplicate lookup (currently once to find the editor view and then again to jump to the correct line).

Reply:
Sounds reasonable to me.